http/rest-client/src/test/java/io/quarkus/ts/http/restclient/GzipMaxInputIT.java

```diff
@@ -0,0 +1,127 @@
+package io.quarkus.ts.http.restclient;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+import jakarta.ws.rs.core.MediaType;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class GzipMaxInputIT {
+
+    final byte[] zero_bytes = new byte[0];
+
+    final String invalid_value = "";
+    final byte[] gzip_bellow_1K_limit_max = new byte[900];
+    final byte[] gzip_same_1K_limit_max = new byte[1024];
+    final byte[] gzip_over_1K_limit_max = new byte[10000000];
+
+    @QuarkusApplication(classes = { GzipResource.class, GzipClientService.class }, properties = "gzip.properties")
+    static RestService app = new RestService();
+
+    /**
+     *
+     * According "All configurations options" guide the property 'quarkus.resteasy.gzip.max-input' refers to
+     *
+     * Maximum deflated file bytes size
+     *
+     * If the limit is exceeded, Resteasy will return Response with status 413("Request Entity Too Large")
+     *
+     */
+
+    @Test
+    void invalidValue() throws IOException {
+        app.given()
+                .contentType(MediaType.TEXT_PLAIN)
+                .header("Content-Encoding", "gzip")
+                .body(invalid_value)
+                .when()
+                .post("/gzip")
+                .then()
+                .statusCode(400);
+    }
+
+    @Disabled("Because https://github.com/quarkusio/quarkus/issues/40806")
+    @Test
+    void sendZeroValue() throws IOException {
+        byte[] compressedData = generateCompressedData(zero_bytes);
+
+        Response response = app.given()
+                .contentType(MediaType.APPLICATION_OCTET_STREAM)
+                .header("Content-Encoding", "gzip")
+                .body(compressedData)
+                .when()
+                .post("/gzip")
+                .then()
+                .extract().response();
+        assertEquals(HttpStatus.SC_INTERNAL_SERVER_ERROR, response.statusCode());
+    }
+
+    @Disabled("Because https://github.com/quarkusio/quarkus/issues/40806")
+    @Test
+    void sendMaxLimit1k() throws IOException {
```

Comment:
Few more things:

Asserts should have a message so when reviewing the failures I don't need to open the source code of the tests to figure out what went wrong

Test method naming is confusing, I have no idea what to expect from them and how they should differ, especially the first two:
 - testGzipOverTheMaxLimit
 - testGzipAboveTheMaxLimit
 - sendMaxLimit1k